nixd/include/nixd/Server/Controller.h

```diff
@@ -122,6 +128,7 @@ class Controller : public lspserver::LSPServer {
 
   std::mutex ConfigLock;
 
+  configuration::TopLevel JSONConfig;
   configuration::TopLevel Config; // GUARDED_BY(ConfigLock)
```

Comment:
Myabe rename this into `LSPWorkspaceConfig`

Reply:
`JSONConfig` is just the config read from `.nixd.json` when the language server starts, and `Config` is the result of `workspace/configuration` with `JSONConfig` used as results. Therefore, I think the names make sense, although maybe `Config` could be changed to a more verbose `LSPWorkspaceWithJSONDefaultsConfig`? However, this may also not be a good name as before `workspace/configuration` is received, `Config` is just set to a copy of `JSONConfig`.